lb_content_resolver/content_resolver.py

```diff
@@ -97,80 +103,78 @@ def traverse(self, relative_path):
 
         return True
 
-    def encode_string(self, text):
-        if text is None:
-            return None
-        return unidecode(re.sub(" +", " ", re.sub(r'[^\w ]+', '', text)).strip().lower())
+    def build_index(self):
+        """
+            Fetch the data from the DB and then build the fuzzy lookup index.
+        """
 
-    def resolve_recording(self, artist_name, recording_name, distance=2):
+        artist_recording_data = []
+        for recording in Recording.select():
+            artist_recording_data.append((recording.artist_name, recording.recording_name, recording.id))
 
-        query = "%s~%d %s~%d" % (self.encode_string(artist_name), distance, self.encode_string(recording_name), distance)
-        self.open_index()
+        self.fuzzy_index = FuzzyIndex(self.index_dir)
+        self.fuzzy_index.build(artist_recording_data)
 
-        ret = []
-        with self.ix.searcher() as searcher:
-            parser = QueryParser("lookup", self.ix.schema)
-            parser.add_plugin(FuzzyTermPlugin())
-            query = parser.parse(query)
-            for result in searcher.search(query):
-                ret.append(dict(result))
+    def encode_string(self, text):
+        """ 
+            Remove unwanted crap from the query string and only keep essential information.
 
-        return ret
+            'This is the ultimate track !!' -> 'thisistheultimatetrack'
+        """
+        if text is None:
+            return None
+        return unidecode(re.sub(" +", " ", re.sub(r'[^\w ]+', '', text)).strip().lower())
```

Comment:
Here be magic ;P

Reply:
Most of it is not too hard to explain. the magic is in unidecode(). If there is anything that makes the mapping intelligent, its is *this* thing. (eg. it can transliterate chinese into an english!)